### PR TITLE
Group Analytics Support

### PR DESCRIPTION
**Pull request by santigracia**
Adding Group Analytics support.

### CODE REVIEW

**Comment by elliotlings** [comment]
I would like to see this supported, too.

File: examples/group_advanced.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// unsets the property test 
+//$mp->group->remove("company","Mixpanel3", array("test"));
+
+
+// add Funnels Cohorts and Flows to a list of "Features" for Group profile Mixpanel3
+//$mp->group->append("company","Mixpanel3", "Features", array("Funnels","Flows","Cohorts") );
+
+
+//deletegroup
+//$mp->group->deleteGroup("company","Mixpanel3");
```
Comment (jaredmixpanel) [inline examples/group_advanced.php:34]: any reason the remove, append and delete examples are commented out?

Comment (jaredmixpanel) [inline examples/group_advanced.php:34]: any reason the remove, append and delete examples are commented out?

File: lib/Mixpanel.php [hunk 2]
```diff
@@ -139,6 +147,7 @@ public function __construct($token, $options = array()) {
         parent::__construct($options);
         $this->people = new Producers_MixpanelPeople($token, $options);
         $this->_events = new Producers_MixpanelEvents($token, $options);
+        $this->group = new Producers_MixpanelGroups($token,$options);
```
Comment (jaredmixpanel) [inline lib/Mixpanel.php:150]:
```suggestion
        $this->group = new Producers_MixpanelGroups($token,$options);
```
```suggestion
        $this->group = new Producers_MixpanelGroups($token, $options);
```

Comment (jaredmixpanel) [inline lib/Mixpanel.php:150]:
```suggestion
        $this->group = new Producers_MixpanelGroups($token,$options);
```
```suggestion
        $this->group = new Producers_MixpanelGroups($token, $options);
```

File: lib/Producers/MixpanelGroups.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * Adds $val to a list located at $prop. If the property does not exist, it will be created. If $val is a string
+     * and the list is empty or does not exist, a new list with one value will be created.
+     * @param string|int $group_key the group_key used for groups in Project Settings
+     * @param string|int $group_id the group id used for the group profile
+     * @param string $prop the property that holds the list
+     * @param string|array $val items to add to the list
+     * @param boolean $ignore_time If the $ignore_time property is true, Mixpanel will not automatically update the "Last Seen" property of the group profile. Otherwise, Mixpanel will add a "Last Seen" property associated with the current time
+     */
+    public function append($group_key, $group_id, $prop, $val, $ignore_time = false) {
```
Comment (jaredmixpanel) [inline lib/Producers/MixpanelGroups.php:78]:
```suggestion
    public function union($group_key, $group_id, $prop, $val, $ignore_time = false) {
```

Comment (jaredmixpanel) [inline lib/Producers/MixpanelGroups.php:78]:
```suggestion
    public function union($group_key, $group_id, $prop, $val, $ignore_time = false) {
```

File: test/Producers/MixpanelGroupsProducerTest.php [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        $this->assertEquals("company", $msg['$group_key']);
+        $this->assertEquals("Mixpanel", $msg['$group_id']);
+        $this->assertEquals("token", $msg['$token']);
+        $this->assertArrayHasKey('$set_once', $msg);
+        $this->assertArrayHasKey("industry", $msg['$set_once']);
+        $this->assertEquals("Tech", $msg['$set_once']['industry']);
+    }
+
+    public function testAppendSingle() {
+        $this->_instance->append("company","Mixpanel", "actions", "Logged In");
```
Comment (jaredmixpanel) [inline test/Producers/MixpanelGroupsProducerTest.php:66]:
```suggestion
        $this->_instance->union("company","Mixpanel", "actions", "Logged In");
```

File: test/Producers/MixpanelGroupsProducerTest.php [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+        $this->assertEquals("company", $msg['$group_key']);
+        $this->assertEquals("Mixpanel", $msg['$group_id']);
+        $this->assertEquals("token", $msg['$token']);
+        $this->assertArrayHasKey('$set_once', $msg);
+        $this->assertArrayHasKey("industry", $msg['$set_once']);
+        $this->assertEquals("Tech", $msg['$set_once']['industry']);
+    }
+
+    public function testAppendSingle() {
```
Comment (jaredmixpanel) [inline test/Producers/MixpanelGroupsProducerTest.php:65]:
```suggestion
    public function testUnionSingle() {
```

Comment (jaredmixpanel) [inline test/Producers/MixpanelGroupsProducerTest.php:65]:
```suggestion
    public function testUnionSingle() {
```

File: test/Producers/MixpanelGroupsProducerTest.php [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        $msg = $queue[count($queue)-1];
+
+        $this->assertEquals("company", $msg['$group_key']);
+        $this->assertEquals("Mixpanel", $msg['$group_id']);
+        $this->assertArrayHasKey('$union', $msg);
+        $this->assertArrayHasKey("actions", $msg['$union']);
+        $this->assertEquals("Logged In", $msg['$union']['actions']);
+    }
+
+    public function testAppendMultiple() {
```
Comment (jaredmixpanel) [inline test/Producers/MixpanelGroupsProducerTest.php:77]:
```suggestion
    public function testUnionMultiple() {
```

File: test/Producers/MixpanelGroupsProducerTest.php [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+        $this->_instance->append("company","Mixpanel", "actions", array("Logged In", "Logged Out"));
```
Comment (jaredmixpanel) [inline test/Producers/MixpanelGroupsProducerTest.php:78]:
```suggestion
        $this->_instance->union("company","Mixpanel", "actions", array("Logged In", "Logged Out"));
```

Comment (jaredmixpanel) [inline test/Producers/MixpanelGroupsProducerTest.php:78]:
```suggestion
        $this->_instance->union("company","Mixpanel", "actions", array("Logged In", "Logged Out"));
```

**Review by jaredmixpanel — COMMENTED** [review]
@santigracia thanks for this! it looks good to me, just a couple small things.

File: examples/group_advanced.php (hunk 1 above)
Comment (santigracia) [reply examples/group_advanced.php:34]: oh, forgot to uncomment, it was to see the results of the set operations after running the test without deleting the group

**Review by jaredmixpanel — APPROVED** [review]
LGTM

File: lib/Producers/MixpanelGroups.php [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+     * Unset properties on a group profile. If the group does not exist, it creates it with no properties.
+     * If it does exist, it unsets these properties. NOTE: In other libraries we use 'unset' which is
+     * a reserved word in PHP.
+     * @param string|int $group_key the group_key used for groups in Project Settings
+     * @param string|int $group_id the group id used for the group profile
+     * @param array $props associative array of properties to unset on the group profile
+     * @param boolean $ignore_time If the $ignore_time property is true, Mixpanel will not automatically update the "Last Seen" property of the group profile. Otherwise, Mixpanel will add a "Last Seen" property associated with the current time
+     */
+    public function remove($group_key, $group_id, $props, $ignore_time = false) {
+        $payload = $this->_constructPayload($group_key, $group_id, '$unset', $props, $ignore_time);
```
Comment (zihejia) [inline lib/Producers/MixpanelGroups.php:65]: shoud be `$remove`

Comment (zihejia) [reply lib/Producers/MixpanelGroups.php:65]: also, add another method for unset

Comment (jaredmixpanel) [inline lib/Producers/MixpanelGroups.php:65]: good catch, @zihejia 
```suggestion
        $payload = $this->_constructPayload($group_key, $group_id, '$remove', $props, $ignore_time);
```

**Review by zihejia — APPROVED** [review]

**Pull request merged**